### PR TITLE
Add shader validation tests about using `blend_src` and enabling extension

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
This patch adds shader validation tests to verify that using `blend_src` the extension `dual_source_blending` must be enabled in the shader.

This patch also rolls @webgpu/types to introduce the newly added `dual-source-blending` feature name.

Issue: #3810

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [*] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/extension/dual_source_blending.spec.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+g.test('use_blend_src_requires_extension_enabled')
+  .desc(
+    `Checks that the blend_src attribute is only allowed with the extension dual_source_blending
+     supported`
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase({ requiredFeatures: ['dual-source-blending'] });
+  })
+  .params(u => u.combine('enableExtension', [true, false] as const))
```
Comment (jzm-intel) [inline src/webgpu/shader/validation/extension/dual_source_blending.spec.ts:18]: I thought we should also test on feature-not-supported devices that not using the feature ends up in validation error.

``` suggestion
  .params(u => u.combine('enableExtension', [true, false] as const))
  .beforeAllSubcases(t => {
    if (t.params.enableExtension) {
      t.selectDeviceOrSkipTestCase({ requiredFeatures: ['dual-source-blending'] });
    }
  })
```

File: src/webgpu/shader/validation/extension/dual_source_blending.spec.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('use_blend_src_requires_extension_enabled')
+  .desc(
+    `Checks that the blend_src attribute is only allowed with the extension dual_source_blending
+     supported`
```
Comment (jzm-intel) [inline src/webgpu/shader/validation/extension/dual_source_blending.spec.ts:13]:
``` suggestion
enabled in shader.`
```

**Review by jzm-intel — APPROVED** [review]
LGTM with nits, thanks!

File: src/webgpu/shader/validation/extension/dual_source_blending.spec.ts (hunk 1 above)
Comment (Jiawei-Shao) [reply src/webgpu/shader/validation/extension/dual_source_blending.spec.ts]: I've updated the test here. PTAL, thanks!

**Review by jzm-intel — APPROVED** [review]
LGTM, thanks!

**Review by jrprice — APPROVED** [review]
LGTM

**Pull request merged**